### PR TITLE
Implement Item.deepcopy()

### LINKED ISSUE #1493: Copying items not work correctly with list values
> > item = Item()
> > item['some_field'] = []
> > 
> > c_item = Item(item)
> > c_item['some_field'].append(1)
> > 
> > print item['some_field'], c_item['some_field']
> > [1] [1]

### PR DESCRIPTION
**Pull request by Gallaecio**
Fixes #1493

### CODE REVIEW

File: docs/topics/items.rst [hunk 1]
```diff
@@ -183,6 +174,33 @@ Creating items from dicts::
...
+
+    >>> product2 = Product(product)
+    >>> print(product2)
+    Product(name='Desktop PC', price=1000, tags=['tag1'])
+
+    >>> product3 = product.copy()
+    >>> print(product3)
+    Product(name='Desktop PC', price=1000, tags=['tag1'])
+
+    >>> product4 = product.deepcopy()
```
Comment (kmike) [inline docs/topics/items.rst]: Hey, what do you think about providing some text describing it, not only an example? It'd be nice to mention caveats of using `Product(product)` or `product.copy()`, and tell where user should prefer which way of copying.

**Comment by Gallaecio** [comment]
I’ve created a separate documentation section just for copying items. There I introduce the concepts of shallow copy and deep copy with an example specific to Scrapy items, and link to the Python documentation for further information. Then I introduce the syntax variantes to copy Scrapy items with inline code examples.

File: scrapy/item.py [hunk 2]
```diff
@@ -96,6 +97,13 @@ def __repr__(self):
     def copy(self):
         return self.__class__(self)
 
+    def deepcopy(self):
+        """Returns a `deep copy`_ of this item.
```
Comment (kmike) [inline scrapy/item.py]: Super-minor nitpick: according to [PEP-257](https://www.python.org/dev/peps/pep-0257/), it should be "Return ...", not "Returns ...".

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

**Comment by kmike** [comment]
Updated docs look awesome, thanks @Gallaecio!

**Comment by kmike** [comment]
Thanks @Gallaecio! AppVeyor failure is not related to this PR, so I'm merging it.

**Pull request merged**